sormas-ui/src/main/java/de/symeda/sormas/ui/events/EventParticipantDataView.java

```diff
@@ -184,24 +184,32 @@ protected void initView(String params) {
 			&& event.getDisease() != null) {
 			if (!FacadeProvider.getFeatureConfigurationFacade()
 				.isPropertyValueTrue(FeatureType.IMMUNIZATION_MANAGEMENT, FeatureTypeProperty.REDUCED)) {
-				final ImmunizationListCriteria immunizationListCriteria =
-					new ImmunizationListCriteria.Builder(eventParticipant.getPerson().toReference()).wihDisease(event.getDisease()).build();
 				layout.addSidePanelComponent(
-					new SideComponentLayout(new ImmunizationListComponent(immunizationListCriteria, this::showUnsavedChangesPopup)),
+					new SideComponentLayout(
+						new ImmunizationListComponent(
+							() -> new ImmunizationListCriteria.Builder(eventParticipant.getPerson().toReference()).withDisease(event.getDisease())
+								.build(),
+							this::showUnsavedChangesPopup)),
 					IMMUNIZATION_LOC);
 			} else {
-				VaccinationListCriteria criteria = vaccinationCriteria.vaccinationAssociationType(VaccinationAssociationType.EVENT_PARTICIPANT)
-					.eventParticipantReference(getReference())
-					.region(eventParticipant.getRegion() != null ? eventParticipant.getRegion() : event.getEventLocation().getRegion())
-					.district(eventParticipant.getDistrict() != null ? eventParticipant.getDistrict() : event.getEventLocation().getDistrict());
-				layout.addSidePanelComponent(
-					new SideComponentLayout(new VaccinationListComponent(criteria, this::showUnsavedChangesPopup)),
-					VACCINATIONS_LOC);
+				layout.addSidePanelComponent(new SideComponentLayout(new VaccinationListComponent(() -> {
+					EventParticipantDto refreshedEventParticipant =
+						FacadeProvider.getEventParticipantFacade().getEventParticipantByUuid(getReference().getUuid());
+					return vaccinationCriteria.vaccinationAssociationType(VaccinationAssociationType.EVENT_PARTICIPANT)
```

Comment:
it's a bit hard to read the return, the extractions of some parts into a method would help

ex: refreshedEventParticipant.getRegion() != null? refreshedEventParticipant.getRegion()
								: event.getEventLocation().getRegion())

or refreshedEventParticipant.getDistrict() != null
								? refreshedEventParticipant.getDistrict()
								: event.getEventLocation().getDistrict());